specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-07-02-preview/managedClusters.json

```diff
@@ -896,6 +896,53 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedclusters/{resourceName}/agentPools/{agentPoolName}/abort": {
+      "post": {
+        "tags": [
+          "AgentPools"
+        ],
+        "operationId": "AgentPools_AbortLatestOperation",
+        "summary": "Aborts last operation running on agent pool.",
+        "description": "Aborting last running operation on agent pool. We return a 204 no content code here to indicate that the operation has been accepted and an abort will be attempted but is not guaranteed to complete successfully. Please look up the provisioning state of the agent pool to keep track of whether it changes to Canceled. A canceled provisioning state indicates that the abort was successful",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "name": "agentPoolName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the agent pool."
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "NoContent"
+          },
```

Comment:
We had a conversation about it with the ARM Review board. We will want to go ahead with option 3 listed on the document with 202 and operation tracking but we decided to go with the 204 for now for preview while we make the 202 change as that will need a lot more changes in the backend since there are several considerations with the operation id tracking.
https://stackoverflow.microsoft.com/questions/315314